futures/tests/connection.rs

```diff
@@ -65,3 +73,144 @@ fn connection() {
+        let id = create_purge_channel.id;
+        info!("created channel with id: {}", id);
+
+        create_purge_channel
+          .queue_declare(
+            "hello_again",
+            &QueueDeclareOptions::default(),
+            &FieldTable::new(),
+          ).and_then(move |_| {
+            // Purge the queue so we don't end up with messages from previous test runs!
+            create_purge_channel.queue_purge(
+              "hello_again",
+              &QueuePurgeOptions::default(),
+            ).map(move |_| {
+              info!("channel {} declared queue {}", id, "hello_again");
+            })
+          })
+      });
+
+      // Create a channel and use it to consume messages.
+      // The result of the future will be a single Delivery.
+      let consume_one_message_future = client.create_channel().and_then(|consume_channel| {
+        let id = consume_channel.id;
+        info!("created channel with id: {}", id);
+        let ack_channel = consume_channel.clone();
+
+        let deadline_when = Instant::now() + Duration::from_millis(1000);
+        consume_channel.basic_consume(
+          "hello_again",
+          "my_consumer",
+          &BasicConsumeOptions::default(),
+          &FieldTable::new(),
+        ).and_then(|stream| {
+          info!("got consumer stream");
+
+          // We've purged the queue, and we're only sending one message below.
+          stream.into_future()
+            .map(|(maybe_message, _stream)| maybe_message.expect("Should have been at least one message..."))
+            .map_err(|(err, _stream)| err)
+        })
+        .and_then(move |message| {
+          info!("got message: {:?}", message);
+          info!("decoded message: {:?}", std::str::from_utf8(&message.data).unwrap());
+          ack_channel.basic_ack(message.delivery_tag).map(move |_| {
+            message
+          })
+        })
+        .deadline(deadline_when)
+        .map_err(|err| {
+          panic!("Didn't receive a message in time: {:?}", err);
+        })
+      });
+
+      // Create channel and use it to publish messages.
+      let publish_future = client.create_channel().and_then(move |publish_channel| {
+        let id = publish_channel.id;
+        info!("created channel with id: {}", id);
+        publish_channel
+          .basic_publish(
+            "",
+            "hello_again",
+            b"please receive me",
+            &BasicPublishOptions::default(),
+            BasicProperties::default()
+              .with_user_id("guest".to_string())
+              .with_reply_to("foobar".to_string()),
+          )
+          .map(|confirmation| {
+            info!(
+              "publish got confirmation: {:?}",
+              confirmation
+            )
+          })
+      });
+
+      // (JP:) Both "Example 1" and "Example 2" below should work,
+      // but don't.
+      //
+      // Prior to some refactoring that I don't believe should
```

Comment:
This might actually be https://github.com/sozu-proxy/lapin/issues/32. So I might need to update this test to declare the queue on all channels.